File: Android/app/src/main/java/com/dkaluta/prosary/ui/shared/CategoryLabels.kt
package com.dkaluta.prosary.ui.shared

import android.content.Context
import com.dkaluta.prosary.R

/** Translate known category captions while keeping manifest tags stable for filtering. */
object CategoryLabels {
    private val resources = mapOf(
        "marian" to R.string.category_marian,
        "daily" to R.string.category_daily,
        "mercy" to R.string.category_mercy,
        "franciscan" to R.string.category_franciscan,
        "advent" to R.string.category_advent,
        "meditative" to R.string.category_meditative,
        "passion" to R.string.category_passion,
        "eastern" to R.string.category_eastern,
        "short" to R.string.category_short,
        "easter" to R.string.category_easter,
        "other" to R.string.category_other,
    )

    fun label(tag: String, context: Context): String = resources[tag]?.let(context::getString)
        ?: tag.replaceFirstChar { it.uppercaseChar() }
}

File: Windows/Prosary/Localization/CategoryLabels.cs
namespace Prosary.Localization;

/// <summary>Manifest tags remain stable identifiers; captions follow the interface language.</summary>
public static class CategoryLabels
{
    public static string Display(string tag) => tag.ToLowerInvariant() switch
    {
        "marian" => Loc.Tr("category_marian", "Marian"),
        "daily" => Loc.Tr("category_daily", "Daily"),
        "mercy" => Loc.Tr("category_mercy", "Mercy"),
        "franciscan" => Loc.Tr("category_franciscan", "Franciscan"),
        "advent" => Loc.Tr("category_advent", "Advent"),
        "meditative" => Loc.Tr("category_meditative", "Meditative"),
        "passion" => Loc.Tr("category_passion", "Passion"),
        "eastern" => Loc.Tr("category_eastern", "Eastern"),
        "short" => Loc.Tr("category_short", "Short"),
        "easter" => Loc.Tr("category_easter", "Easter"),
        "other" => Loc.Tr("category_other", "Other"),
        _ => string.IsNullOrEmpty(tag) ? string.Empty : char.ToUpper(tag[0]) + tag[1..],
    };
}
